rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -552,6 +560,7 @@ bool Rir2Pir::compileBC(
     case Opcode::movloc_:
     case Opcode::isobj_:
     case Opcode::check_missing_:
+        log.unsupportedBC("Unsupported BC (are you recompiling?)", bc);
         assert(false && "Recompiling PIR not supported for now.");
```

Comment:
Replace the asserts with just signalling failure?

Reply:
I think we still want this to fail, because how the compiler is invoked, it should actually not happen currently that we recompile pir optimized code